cleverhans/model.py

```diff
@@ -67,3 +67,28 @@ def fprop(self, x):
                  representation of their output.
         """
         raise NotImplementedError('`_fprop` not implemented.')
+
+
+class CallableModelWrapper(Model):
+    def __init__(self, callable_fn, output_layer):
+        """
+        Wrap a callable function that takes a tensor as input and returns
+        a tensor as output with the given layer name.
+        :param callable_fn: The callable function taking a tensor and
+                            returning a given layer as output.
+        :param output_layer: A string of either "probs" or "logits",
+                             corresponding to the output of the callable.
+        """
```

Comment:
Do we need to enforce this? We could imagine maybe making a CallableModelWrapper for producing the features for a Sara Sabour-style attack, where we'd want the output to have a different name. If we don't enforce this here, there will still be an error message later saying that the Model doesn't provide the desired layer.